app/src/main/java/Fragments/AddItemFragment.java

```diff
@@ -85,89 +87,94 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            @Override
+            public void onClick(View v) {
+               //Set camera click listener
+            }
+        });
+
+        selectImageGalleryLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                Intent intent = new Intent(Intent.ACTION_PICK);
+                intent.setType("image/*");
+                startActivityForResult(intent, GALLERY_INTENT);
+            }
+        });
+
+        postItemButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                AlertDialog.Builder builder1 = new AlertDialog.Builder(getContext());
+                builder1.setCancelable(true);
+
+                builder1.setPositiveButton(
+                        "OK",
+                        new DialogInterface.OnClickListener() {
+                            public void onClick(DialogInterface dialog, int id) {
+                                isItemPostable = false;
+                                dialog.cancel();
+                            }
+                        });
+
+                if (isEmpty(itemNameET)) {
+                    builder1.setMessage("Please enter a name for your item!");
+                    AlertDialog alert11 = builder1.create();
+                    alert11.show();
+
+                } else if (isEmpty(itemPriceET)) {
+                    builder1.setMessage("Please enter a price for your item!");
+                    AlertDialog alert11 = builder1.create();
+                    alert11.show();
+
+                } else if (isEmpty(itemDescriptionET)) {
+                    builder1.setMessage("Please enter a description for your item!");
+                    AlertDialog alert11 = builder1.create();
+                    alert11.show();
+                } else {
+                    isItemPostable = true;
+                }
+
+                if (isItemPostable) {
+                    String itemName = itemNameET.getText().toString();
+                    double itemPrice = Double.parseDouble(itemPriceET.getText().toString());
+                    String itemDescription = itemDescriptionET.getText().toString();
+
+                    String id = UUID.randomUUID().toString();
+                    ItemDescription listingItem = new ItemDescription(id, itemName, itemPrice, "", itemDescription);
+
+                    databaseItems.child(id).child("Price").setValue(itemPrice);
+                    databaseItems.child(id).child("Description").setValue(itemDescription);
+                    databaseItems.child(id).child("Name").setValue(itemName);
+                    databaseItems.child(id).child("ImageURL").setValue("");
+                    databaseItems.child(id).child("OwnerID").setValue("userUID");
+
+                    Toast.makeText(getContext(), "Item Posted", Toast.LENGTH_LONG).show();
+
```

Comment:
Would we be able to redirect the user to the home listings page once the adding item function is completed? Though I do like the notification which tells the function of adding is complete!

Reply:
So worst case we can make the tweak (redirecting to listing page after adding an item) later!